{"url": "https://api.github.com/repos/rust-lang/rust/issues/43837", "repository_url": "https://api.github.com/repos/rust-lang/rust", "labels_url": "https://api.github.com/repos/rust-lang/rust/issues/43837/labels{/name}", "comments_url": "https://api.github.com/repos/rust-lang/rust/issues/43837/comments", "events_url": "https://api.github.com/repos/rust-lang/rust/issues/43837/events", "html_url": "https://github.com/rust-lang/rust/issues/43837", "id": 249860366, "node_id": "MDU6SXNzdWUyNDk4NjAzNjY=", "number": 43837, "title": "return x; and x not equivalent at end of function?", "user": {"login": "lheckemann", "id": 341954, "node_id": "MDQ6VXNlcjM0MTk1NA==", "avatar_url": "https://avatars.githubusercontent.com/u/341954?v=4", "gravatar_id": "", "url": "https://api.github.com/users/lheckemann", "html_url": "https://github.com/lheckemann", "followers_url": "https://api.github.com/users/lheckemann/followers", "following_url": "https://api.github.com/users/lheckemann/following{/other_user}", "gists_url": "https://api.github.com/users/lheckemann/gists{/gist_id}", "starred_url": "https://api.github.com/users/lheckemann/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/lheckemann/subscriptions", "organizations_url": "https://api.github.com/users/lheckemann/orgs", "repos_url": "https://api.github.com/users/lheckemann/repos", "events_url": "https://api.github.com/users/lheckemann/events{/privacy}", "received_events_url": "https://api.github.com/users/lheckemann/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 5, "created_at": "2017-08-13T08:31:53Z", "updated_at": "2018-02-26T08:27:06Z", "closed_at": "2017-08-13T08:56:32Z", "author_association": "NONE", "active_lock_reason": null, "body": "I have [some code which the borrow checker doesn't agree with](https://play.rust-lang.org/?gist=ea9e2492053ebb69ef80915500d696e3&version=stable). However, it works fine if I [replace the expression at the end of the function with a return statement containing the same expression](https://play.rust-lang.org/?gist=ea9e2492053ebb69ef80915500d696e3&version=stable). I believed that `return x;` was fully equivalent to `x` at the end of a function, but apparently not? Not sure if this is a documentation issue or a compiler bug, but after raising this on #rust-beginners it was recommended that I raise it here :)", "closed_by": {"login": "arielb1", "id": 1830974, "node_id": "MDQ6VXNlcjE4MzA5NzQ=", "avatar_url": "https://avatars.githubusercontent.com/u/1830974?v=4", "gravatar_id": "", "url": "https://api.github.com/users/arielb1", "html_url": "https://github.com/arielb1", "followers_url": "https://api.github.com/users/arielb1/followers", "following_url": "https://api.github.com/users/arielb1/following{/other_user}", "gists_url": "https://api.github.com/users/arielb1/gists{/gist_id}", "starred_url": "https://api.github.com/users/arielb1/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/arielb1/subscriptions", "organizations_url": "https://api.github.com/users/arielb1/orgs", "repos_url": "https://api.github.com/users/arielb1/repos", "events_url": "https://api.github.com/users/arielb1/events{/privacy}", "received_events_url": "https://api.github.com/users/arielb1/received_events", "type": "User", "site_admin": false}, "reactions": {"url": "https://api.github.com/repos/rust-lang/rust/issues/43837/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/rust-lang/rust/issues/43837/timeline", "performed_via_github_app": null, "state_reason": "completed"}